runtime/tr.source/trj9/z/codegen/J9S390CHelperLinkage.cpp

```diff
@@ -390,41 +388,50 @@ TR::Register * TR::S390CHelperLinkage::buildDirectDispatch(TR::Node * callNode,
 		}
 	else
 		{
-      // Fastpath helper do not expects GC call in-between so only attaching them for normal dual mode helpers 
-		cursor->setNeedsGCMap(0x0000FFFF);
+      // Fastpath helper do not expects GC call in-between so only attaching them for normal dual mode helpers
+      // As GC map is attached to instruction after RA is done, it is guaranteed that all the non-preserved register by system linkage are either stored in preserved register
+      // Or spilled to stack. We only need to mark preserved register in GC map. Only possiblity of non-preserved register containing a live object is in argument to helper which should be a clobberable copy of actual object. 
+		cursor->setNeedsGCMap(getPreservedRegisterMapForGC());
```

Comment:
You seem to be using a mix of tabs and spaces in this commit. Please use spaces only as the rest of the codebase does.